### ISSUE TITLE
Syntax highlighting Liquid tags are not working in feed entry

### ISSUE THREAD

**Issue by MilanAryal** [jekyll/jekyll-feed#58]
Jekyll Pygments syntax highlighting Liquid tags are not rendering a code block in feed entry.

Following are issues feed examples generated by `jekyll-feed` plugin:

### Example 1

![capture-ben-feed](https://cloud.githubusercontent.com/assets/9361180/8200677/3688c3da-14e8-11e5-8ab1-669613fcbe38.PNG)

#### Example 1 — Feed output

![capture-ben-feed-output](https://cloud.githubusercontent.com/assets/9361180/8201613/5c6fd64c-14f2-11e5-98b3-38c82c493d79.PNG)

### Example 2

![capture-milan-feed](https://cloud.githubusercontent.com/assets/9361180/8200679/3bbc6578-14e8-11e5-9c42-1cf4fea9fe85.PNG)

When I try to test/check the issue in the post excerpt:
![capture-milan-post-excerpt](https://cloud.githubusercontent.com/assets/9361180/8202768/9f866c70-14fc-11e5-8eeb-7550252ebedc.PNG)

#### Example 2 — Feed output

![capture-milan-feed-output](https://cloud.githubusercontent.com/assets/9361180/8201458/0a6116b4-14f1-11e5-8264-c5272137b1f1.PNG)

---

Not sure what's the main reason behind this issue but when I used to have custom Atom feed template for Jekyll, it used to work with the following pattern:

``` liquid
{{ post.content | xml_escape }}
```

``` liquid
{{ post.excerpt | strip_html | strip_newlines | xml_escape }}
```

I found `jekyll-feed` plugin using following pattern, where `markdownify` may have created this issue:

``` liquid
{{ post.content | markdownify | xml_escape }}
```

``` liquid
{{ post.excerpt | markdownify | strip_html | strip_newlines | xml_escape }}
```

**Note:** But with the same pattern **there is no problem in HTML output** like in meta description with the following code using `markdownify`:

``` liquid
{{ page.excerpt | markdownify | strip_html | strip_newlines | truncate: 160 }} 
```

**Comment by pathawks** [comment]
This is a known issue.

Posts are not fully rendered before being included in the feed. Using the `markdownify` filter is a hack until we can figure out how to defer generating the feed until after the posts have been fully rendered.

Perhaps a note in the docs would be useful in the meantime?

**Comment by parkr** [comment]
> Perhaps a note in the docs would be useful in the meantime?

:+1:

> Posts are not fully rendered before being included in the feed. Using the markdownify filter is a hack until we can figure out how to defer generating the feed until after the posts have been fully rendered.

We can detect Jekyll 3.0 and if it's Jekyll 3, use a hook instead of a generator.

**Comment by maul-esel** [comment]
Hi! I stumbled about a similar problem just now, using the `{% post_url %}` tag. What do you think of the following interim solution (until jekyll 3):

```ruby
module Jekyll
  module LiquifyFilter
    def liquify(input)
      Liquid::Template.parse(input).render(@context)
    end
  end
end

Liquid::Template.register_filter(Jekyll::LiquifyFilter)
```
(see http://stackoverflow.com/questions/14487110/include-jekyll-liquid-template-data-in-a-yaml-variable)

and in `feed.xml`:
```xml
<content type="html" xml:base="{{ post.url | prepend: url_base | xml_escape }}">{{ post.content | liquify | markdownify | xml_escape }}</content>
```
and
```xml
<summary>{{ post.excerpt | liquify | markdownify | strip_html | strip_newlines | xml_escape }}</summary>
```

**Comment by leo** [comment]
Could someone please fix this? There are many people out there who just started using this plugin on their GitHub Pages setup and are now annoyed by this bug.

Isn't parsing markup a fundamental thing before spitting out the content? :blush:

**Comment by parkr** [comment]
@leo This is not that cut-and-dry, in fact. Please see: https://github.com/jekyll/jekyll-feed/issues/61#issuecomment-141707013

**Comment by pathawks** [comment]
This should be fixed by #73
